Generator/Sources/NeedleFramework/Generating/DependencyProviderSerializerTask.swift

```diff
@@ -58,31 +58,51 @@ class DependencyProviderSerializerTask: AbstractTask<[SerializedProvider]> {
         var result = [SerializedProvider]()
         let (baseClass, content) = serializedBase(for: providers.first!)
         if providers.first?.isEmptyDependency == false {
-            result.append(SerializedProvider(content: content, registration: ""))
+            result.append(SerializedProvider(content: content, registration: "", attributes: [:]))
         }
         for provider in providers {
-            let content = provider.isEmptyDependency ? "" : serializedContent(for: provider, baseClassSerializer: baseClass)
-            let registration = DependencyProviderRegistrationSerializer(provider: provider).serialize()
-            result.append(SerializedProvider(content: content, registration: registration))
+            let paramsSerializer = DependencyProviderParamsSerializer(provider: provider)
```

Comment:
I think I saw this code block (or roughly this) repeated a lot; maybe you can write a protocol to do this in an extension to reuse code?